src/Elmish.WPF/BindingInternal.fs

```diff
@@ -4,8 +4,41 @@ open System.Collections.ObjectModel
 open System.Windows
 
 open Elmish
+open Microsoft.Extensions.Logging
 
 
+type LoggingViewModelArgs =
+  { performanceLogThresholdMs: int
+    log: ILogger
+    logPerformance: ILogger
+    nameChain: string }
+
+module LoggingViewModelArgs =
+
+  let getNameChainFor nameChain name =
+    sprintf "%s.%s" nameChain name
+
+  let getNameChainForItem nameChain collectionBindingName itemId =
+    sprintf "%s.%s.%s" nameChain collectionBindingName itemId
+
+  let map nameChain v = { v with nameChain = nameChain }
+
+type ViewModelArgs<'model, 'msg> =
+  { initialModel: 'model
+    dispatch: 'msg -> unit
+    loggingArgs: LoggingViewModelArgs }
+
+module ViewModelArgs =
+  let create initialModel dispatch nameChain loggingArgs =
+    { initialModel = initialModel
+      dispatch = dispatch
+      loggingArgs = LoggingViewModelArgs.map nameChain loggingArgs }
+
+  let map mapModel mapMsg v =
+    { initialModel = v.initialModel |> mapModel
+      dispatch = mapMsg >> v.dispatch
+      loggingArgs = v.loggingArgs }
+
```

Comment:
Why moving these here?  Just to have less things in `ViewModel.fs`?  I don't think they belong in this file.  Maybe we should create a new file for them.

Reply:
Yes it's intentional. They are used in the `BindingsInternal` file to define the args that view models get when they are created.